integration-tests/stable/select.test.ts

```diff
@@ -0,0 +1,58 @@
+import type * as edgedb from "edgedb";
+
+import e, { type $infer } from "./dbschema/edgeql-js";
+import { setupTests, teardownTests, tc } from "./setupTeardown";
+
+let client: edgedb.Client;
+
+describe("select", () => {
+  beforeAll(async () => {
+    const setup = await setupTests();
+    ({ client } = setup);
+  });
+
+  afterAll(async () => {
+    await teardownTests(client);
+  }, 10_000);
+
+  test("union of polymorphic types in computed schema field", () => {
+    const WatchingList = e.shape(e.User.watching_list, () => ({
+      index: true,
+      title: true,
+      ...e.is(e.Movie, {
+        plot: true,
+      }),
+      ...e.is(e.Show, {
+        seasons: true,
+      }),
+    }));
```

Comment:
I think we should have an expect error test trying to select a field that only exists on one of the union items outside the `e.is(...)` to check we reject it.